src/test/isolation2/input/alter_db_set_tablespace.source

```diff
@@ -0,0 +1,81 @@
+-- If a mirror checkpointer doesn't create checkpoint between making changes in
+-- a database and moving this database to another tablespace, then attempt to
+-- create checkpoint the next time should not result in the fsync error.
+
+
+-- start_matchsubs
+
+-- # Remove source file name and line
+-- m/^ERROR:  failed to inject fault: ERROR:  fault not set, fault name:'\(null\)' \(.*\)/
+-- s/^ERROR:  failed to inject fault: ERROR:  fault not set, fault name:'\(null\)' \(.*\)/ERROR:  failed to inject fault: ERROR:  fault not set, fault name:'(null)' (file:line)/
+
+-- end_matchsubs
+
+
+-- Setup
+!\retcode  mkdir -p @testtablespace@/adst_dest;
+-- Set fsync on since we need to test the fsync code logic.
+!\retcode gpconfig -c fsync -v on --skipvalidation;
+!\retcode gpstop -u;
+CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
+
+CREATE OR REPLACE FUNCTION mirror0() RETURNS gp_segment_configuration.dbid%type
+AS $$
+    SELECT dbid FROM gp_segment_configuration WHERE content = 0 AND role = 'm'
+$$ LANGUAGE SQL EXECUTE ON MASTER;
+
+-- Checkpointer must not start processing requests until database is moved
+-- to another tablespace
+SELECT gp_inject_fault('ckpt_loop_begin', 'infinite_loop', mirror0());
+
+-- Wait until checkpointer is ready to start processing requests
+SELECT gp_wait_until_triggered_fault('ckpt_loop_begin', 1, mirror0());
```

Comment:
Maybe temporary decrease `checkpoint_timeout` to speed up test?

Reply:
> Maybe temporary decrease `checkpoint_timeout` to speed up test?

`checkpoint_timeout` can not be changed, because it has GUC_DISALLOW_USER_SET flag

Reply:
Does it really prevent from changing with postgresql.conf (gpconfig) similar that is already done by your code?

Reply:
Does it really prevent from changing in postgresql.conf (gpconfig), similar that is already done by your changes.

Reply:
ok. I've seen. I was confused by GUC description in doc.  Thank you.